test/compile/mark/area.test.ts

```diff
@@ -49,4 +49,38 @@ describe('Mark: Area', function() {
       assert.deepEqual(props.fill, {scale: COLOR, field: 'Miles_per_Gallon'});
     });
   });
+  
```

Comment:
Good that you add new tests for new cases.  However, it seems like we don't have test to cover old cases yet and you're changing the logic for old cases in this PR and we should prevent regression.  

Can you checkout the main branch, write unit test for other cases in areas. (use `npm run cover` to see coverage in area.ts), then merge those tests into this branch?  This way we can make sure that we don't introduce new bug in this branch.  :)